### DISCUSSION TITLE
How to use this library with Remix Vite setup?

### DISCUSSION THREAD

**Discussion by AdiRishi** [abereghici/remix-themes#38] [Q&A]
There are a few core changes in the new Remix templates that make it quite unclear how to set this project up.

Specifically, they  revolve around the new [Root Layout](https://remix.run/docs/en/main/file-conventions/root#layout-export) component.
The `root.tsx` file for a new remix project now looks like this
```ts
import {
  Links,
  Meta,
  Outlet,
  Scripts,
  ScrollRestoration,
} from "@remix-run/react";

export function Layout({ children }: { children: React.ReactNode }) {
  return (
    <html lang="en">
      <head>
        <meta charSet="utf-8" />
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <Meta />
        <Links />
      </head>
      <body>
        {children}
        <ScrollRestoration />
        <Scripts />
      </body>
    </html>
  );
}

export default function App() {
  return <Outlet />;
}
```

So a few questions arise
- It is not possible to call `useLoaderData` inside the Layout component, so how do we load the necessary data
- The current docs assume that the `App` component is the top layer, moreover, it assumes that we can wrap the app shell in a provider, which is not the case anymore. How do we deal with these changes?

**Comment by abereghici** [comment]
I'm working on a new version to support these changes. 

Meanwhile, we could not export the Layout, but wrap the App, ErrorBoundary and HydrateFallback with it.

I created an example here:
https://github.com/abereghici/vite-remix-themes/blob/main/app/root.tsx
https://github.com/abereghici/vite-remix-themes/blob/main/app/routes/counter.tsx
